systemvm/patches/debian/config/opt/cloud/bin/cs/CsConfig.py

```diff
@@ -68,8 +68,11 @@ def get_domain(self):
 
     def get_dns(self):
         dns = []
-        # Check what happens with use_ext_dns
-        dns.append(self.address().get_guest_ip())
+        if not self.cl.get_use_ext_dns():
+            if not self.is_vpc() and self.cl.is_redundant():
+                dns.append(self.cl.get_guest_gw())
+            elif not self.cl.get_use_ext_dns():
```

Comment:
the exression (not self.cl.get_use_ext_dns()) already known to be true due to line 71, so this check is redundant.